src/08_ecosystem/00_chapter.md

```diff
@@ -0,0 +1,51 @@
+# The Async Ecosystem
+Most Rust asynchronous projects will need third-party crates to function.
+The `futures` crate contains traits and functions useful for writing async code.
+In addition, there are several crates that provide async runtimes,
+which are used to run asynchronous functions from synchronous code.
+This section introduces the crates most commonly used in async projects.
+
+## Asynchronous Runtimes
+Async runtimes are libraries that allow async functions to be run from sync functions, including the `main` function.
+Runtimes usually bundle together a *reactor* with one or more *executors*.
+Reactors can be thought of as event queues.
```

Comment:
The reactor is responsible for async IO and timers, not tasks which is suggested in this paragraph (AFAIU - we don't use this distinction in Fuchsia). I think it's also worth pointing out that futures that interact with async IO or timers are tied to a specific reactor implementation. This is a main source of the lack of runtime interoperability.

Reply:
Can you explain more what it means for the reactor to be responsible for async IO and timers? My mental picture of this was "there is a task to read from a file, but something else has a lock on the file; the executor puts the task to sleep, and the reactor notifies the executor when the lock is released or when the task can make progress again" or something along those lines. I'm now realizing that's not quite correct but I'm not sure what a more accurate description would be.

Reply:
Leaving open for offline chat.